### PR TITLE
Add CI via GitHubAction

### LINKED ISSUE #32: Run tests in CI
...via a GitHubAction.

If this is something you're okay with, @SamuraiAku, I'd be happy to add it.

### PR DESCRIPTION
**Pull request by hannahilea**
Close #32.

- Add GHA that runs CI tests
- Uploads code coverage via codecov
- Add badges to README

... to enable codecov, @SamuraiAku will need to get a codecov account and grant access to this github repository: https://docs.codecov.com/docs/github-2-getting-a-codecov-account-and-uploading-coverage#create-a-codecov-account  This is free for public repositories; if for whatever reason you don't want to enable it, we can remove the codecov section from this PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # SPDX.jl
 
+[![GitHub Actions](https://github.com/SamuraiAku/SPDX.jl/workflows/CI/badge.svg)](https://github.com/SamuraiAku/SPDX.jl/actions/workflows/CI.yml)
+[![codecov](https://codecov.io/gh/SamuraiAku/SPDX.jl/branch/master/graph/badge.svg?token=TODO)](https://codecov.io/gh/SamuraiAku/SPDX.jl)
```
Comment (hannahilea) [inline README.md:4]: Once you've granted codecov access to this repo, you'll be able to replace this "TODO" with the token found at https://app.codecov.io/gh/SamuraiAku/SPDX.jl/settings/badge

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
```
Comment (hannahilea) [inline .github/workflows/CI.yml:5]:
```suggestion
      - master
```

**Comment by SamuraiAku** [comment]
@hannahilea Thanks so much for putting this together!

I'm completely new to this and I want to make sure I comprehend what's going on, so merging this may take a little time.  But this is cool.

**Comment by SamuraiAku** [comment]
The minimum version that we can test with is 1.8, since SPDX.jl uses the `mutable struct` feature. If I understand the YML file correctly, it tries to test with the LTS version v1.6

**Comment by hannahilea** [comment]
Updated to test on 1.8 and 1

**Comment by SamuraiAku** [comment]
Signed up for codecov and ran the action. Showing overall coverage of 54.55%. No coverage in the write functions of course. Other tests need some more work to improve coverage. Stuff to work on over time.

**Pull request merged**